File: gblog-api/app/models/comment.py
from sqlalchemy import Column, String, Text, DateTime, Integer, ForeignKey, BigInteger
from sqlalchemy.dialects.mysql import BIT
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database.database import Base

class Comment(Base):
    __tablename__ = "comment"
    
    id = Column(BigInteger, primary_key=True, autoincrement=True)
    nickname = Column(String(255), nullable=False, comment="昵称")
    email = Column(String(255), nullable=False, comment="邮箱")
    qq = Column(String(255), comment="如果评论昵称为QQ号，则将昵称和头像置为QQ昵称和QQ头像，并将此字段置为QQ号备份")
    avatar = Column(String(255), nullable=False, comment="头像(图片路径)")
    ip = Column(String(255), comment="评论者ip地址")
    website = Column(String(255), comment="个人网站")
    content = Column(String(255), nullable=False, comment="评论内容")
    create_time = Column(DateTime, comment="评论时间")
    is_published = Column(BIT(1), nullable=False, comment="公开或回收站")
    is_admin_comment = Column(BIT(1), nullable=False, comment="博主回复")
    page = Column(Integer, nullable=False, comment="0普通文章，1关于我页面，2友链页面")
    is_notice = Column(BIT(1), nullable=False, comment="接收邮件提醒")
    blog_id = Column(BigInteger, ForeignKey("blog.id"), comment="所属的文章")
    root_comment_id = Column(BigInteger, nullable=False, comment="所属根评论id，顶级评论为-1")
    parent_comment_id = Column(BigInteger, nullable=False, comment="父评论id，-1为根评论")
    
    # 关系
    # blog = relationship("Blog", back_populates="comments")
    # replies = relationship("Comment", backref="parent", remote_side=[id])

File: GBlog-view-backend/src/main/java/fun/wuziran/gblog/viewbackend/model/entity/Comment.java
package fun.wuziran.gblog.viewbackend.model.entity;

import lombok.Data;

/**
 * @Description 评论entity
 * @Author Geralt
 * @Date 2025/1/10
 */
@Data
public class Comment {
    // 评论id
    private long id;
    // 评论者昵称
    private String nickname;
    // 评论者邮箱
    private String email;
    // 评论者QQ号
    private long qq;
    // 评论者头像
    private String avatar;
    // 评论时间
    private String createTime;
    // 评论者网站
    private String website;
    // 评论者IP
    private String ip;
    // 评论内容
    private String content;
    // 是否公开
    private boolean isPublished;
    // 是否是博主评论
    private boolean adminComment;
    // 评论页面
    private int page;
    // 是否接收邮件提醒
    private boolean isNotice;
    // 父评论id
    private long parentCommentId;
    // 根评论id
    private String rootCommentId;
    // 所属博客id
    private long blogId;
}